src/badge/withBadge.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import { Platform, StyleSheet, View } from "react-native";
+import Badge from "./Badge";
+
+const styles = StyleSheet.create({
+  badge: {
+    borderRadius: 9,
+    height: 18,
+    minWidth: 0,
+    width: 18
+  },
+  badgeContainer: {
+    position: "absolute",
+    shadowColor: "black",
+    shadowOffset: {
+      width: 0,
+      height: 0.5
+    },
+    shadowRadius: 0.5,
+    shadowOpacity: 1.0
+  },
+  badgeText: {
+    fontSize: 10,
+    paddingHorizontal: 0
+  }
+});
+
+const withBadge = (value, options = {}) => WrappedComponent =>
+  class extends React.Component {
+    render() {
+      const {
+        status = "error",
+        left = 10,
+        bottom = 15,
+        top = 0,
+        right = 0,
```

Comment:
If you expect left 10 and bottom 15 to put the badge on the top right hand corner, you should use top and right to not worry about the wrapped component dimension